tests/end_to_end_tests/managed_packages_tests/conftest.py

```diff
@@ -0,0 +1,194 @@
+import dataclasses
+import os
+import pathlib as pl
+import shutil
+import subprocess
+import argparse
+import http.server
+import socketserver
+import tarfile
+import threading
+import time
+from typing import List
+
+import pytest
+
+from agent_build_refactored.tools.constants import SOURCE_ROOT
+from agent_build_refactored.tools.runner import Runner, RunnerMappedPath
+from agent_build_refactored.managed_packages.managed_packages_builders import ALL_MANAGED_PACKAGE_BUILDERS, PREPARE_TOOLSET_GLIBC_X86_64, PYTHON_PACKAGE_NAME, AGENT_LIBS_PACKAGE_NAME
+from tests.end_to_end_tests.run_in_remote_machine import DISTROS
+
+
+def pytest_addoption(parser):
+    parser.addoption(
+        "--builder-name",
+        dest="builder_name",
+        required=True
+    )
+
+    parser.addoption(
+        "--packages-source",
+        dest="packages_source",
+        required=False,
+        help="Directory to packages to test. If not specified, packages will be built inplace."
```

Comment:
Package source would be a directory or url (for a repo) right?
Should change the help text to be "Directory or tarball or repo url of packages to test. ..."

If repos are always specified as urls, can eliminate the --packages-source-type option (but up to you of course)